Library/brew.sh

```diff
@@ -207,6 +173,56 @@ EOS
   esac
 fi
 
+if [[ -z "$HOMEBREW_DEVELOPER" ]]
+then
+  unset HOMEBREW_RUBY_PATH
+fi
+
+if [[ -z "$HOMEBREW_RUBY_PATH" && "$HOMEBREW_COMMAND" != "install-vendor" ]]
+then
+  HOMEBREW_VENDOR_RUBY_PATH="$HOMEBREW_LIBRARY/Homebrew/vendor/ruby/opt/bin/ruby"
+  if [[ -x "$HOMEBREW_VENDOR_RUBY_PATH" ]]
+  then
+    HOMEBREW_RUBY_PATH="$HOMEBREW_VENDOR_RUBY_PATH"
+  else
+    if [[ -n "$HOMEBREW_OSX" ]]
+    then
+      HOMEBREW_RUBY_PATH="/System/Library/Frameworks/Ruby.framework/Versions/Current/usr/bin/ruby"
+    else
+      HOMEBREW_RUBY_PATH="$(which ruby)"
```

Comment:
I don't know; I think the majority of cases I've seen a missing system Ruby have been when people have used an official Ruby installer, `rbenv` or `rvm` and deleted it. I think we should try to let people use those tools instead of requiring our vendored Ruby if possible. Ideally I'd even add a `.ruby-version` to the `HOMEBREW_REPOSITORY` root which would ensure those tools pick the "right" Ruby. Otherwise we're not letting people have an option to build from source, really.